src/index.ts

```diff
@@ -119,7 +145,9 @@ interface WeatherResponse {
 const runPlugin = async (e: { uuid: string }) => {
   const query = (await logseq.Editor.getBlock(e.uuid))?.content.split('\n')[0]
 
-  const coords = parseQuery(query || logseq.settings?.localCoordinates)
+  const coords =
+    parseQuery(query || logseq.settings?.localCoordinates) ||
+    parseQuery(logseq.settings?.localCoordinates)
```

Comment:
I'm a bit confused by this - why do we need to call potentially call `parseQuery` twice, using one of the same arguments as the first call?

Reply:
It is because you are grabbing the query parameter with [this](https://github.com/axelav/logseq-plugin-weather/blob/07ab79c01a2606e1855888ed58b32e0ec0742a84/src/index.ts#L120) function.

When I ran the plugin with my example template.

```
- 44.590959, -104.698514
    {{renderer :addCurrentWeatherData}} 
```

the value of `query` was `{{renderer :addCurrentWeatherData}}` during execution and that failed at the [parseQuery](https://github.com/axelav/logseq-plugin-weather/blob/07ab79c01a2606e1855888ed58b32e0ec0742a84/src/index.ts#L87) method. 

Changing the line to 

```
parseQuery(query || logseq.settings?.localCoordinates) ||
    parseQuery(logseq.settings?.localCoordinates)
``` 

Re-tries the parseQuery method explicitly with the localCoordinates from settings if the first ParseQuery failed.

Reply:
Interesting - perhaps I'll try to clean that up sometime. Thanks for explanation!